### PR TITLE
Verify signature at runtime

### LINKED ISSUE #424: Add signature verification in node/runtime
Currently we only verify TX signatures at rpc level. We probably should do this at runtime.
the good place for this is in "fn apply_signed_transaction(" in node/runtime/src/lib.rs after we fetched sender's account and have both signed transaction and public keys.
It requires fixing tests since tests use default signatures.
@azban Why do you need caching?

### PR DESCRIPTION
**Pull request by bowenwang1996**
Fixes #424

### CODE REVIEW

File: node/runtime/src/lib.rs [hunk 1]
```diff
@@ -236,6 +237,12 @@ impl Runtime {
                         sender.nonce,
                     ));
                 }
+                if !verify_transaction_signature(&transaction.clone(), &sender.public_keys) {
```
Comment (evgenykuzyakov) [inline node/runtime/src/lib.rs]: Why clone?

Comment (bowenwang1996) [reply node/runtime/src/lib.rs]: Removed

**Comment by bowenwang1996** [comment]
Removing state is a separate pull request #532 I think github does not update the history properly.

**Comment by bowenwang1996** [comment]
Looks like there's something wrong with CI. Have checked locally that everything works.

**Pull request merged**